server/services/gameLogicClass.js

```diff
@@ -0,0 +1,221 @@
+const cardManager = require('./cards');
+const socket = require('./socket');
+
+module.exports = class GameLogic {
+    constructor(roomId) {
+        this.statusTypes = {
```

Comment:
I think statusTypes can be defined outside of the class as it is essentially an enum